language/move-prover/bytecode-to-boogie/tests/driver/mod.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use bytecode_source_map::source_map::{ModuleSourceMap, SourceMap};
+use bytecode_to_boogie::translator::BoogieTranslator;
+use bytecode_verifier::VerifiedModule;
+use goldenfile;
+use ir_to_bytecode::{compiler::compile_module, parser::ast::Loc, parser::parse_module};
+use libra_tools::tempdir::TempPath;
+use libra_types::account_address::AccountAddress;
+use prettydiff::{basic::DiffOp, diff_lines};
+use std::{env, fs, fs::File, io::Error, io::Read, io::Write, path::Path, process::Command};
+use stdlib::{stdlib_modules, stdlib_source_map};
+
+fn compile_files(file_names: Vec<&str>) -> (Vec<VerifiedModule>, SourceMap<Loc>) {
+    let mut verified_modules = stdlib_modules().to_vec();
+    let mut source_maps = stdlib_source_map().to_vec();
+    let files_len = file_names.len();
+    let dep_files = &file_names[0..files_len];
+
+    // assuming the last file is a program that might contain a script
+    let address = AccountAddress::default();
+    for file_name in dep_files {
+        let code = fs::read_to_string(file_name).unwrap();
+        let module = parse_module(&code).unwrap();
+        let (compiled_module, source_map) =
+            compile_module(address, module, &verified_modules).expect("module failed to compile");
+        let verified_module_res = VerifiedModule::new(compiled_module);
+
+        match verified_module_res {
+            Err(e) => {
+                panic!("{:?}", e);
+            }
+            Ok(verified_module) => {
+                verified_modules.push(verified_module);
+                source_maps.push(source_map);
+            }
+        }
+    }
+    (verified_modules, source_maps)
+}
+
+#[allow(unused_variables)]
+pub fn generate_boogie(file_name: &str, target_modules: &[&str]) -> String {
+    let mut file_names = vec![];
+    file_names.push(file_name);
+    let (modules, source_maps) = compile_files(file_names.to_vec());
+
+    if cfg!(feature = "golden") {
```

Comment:
For my edification, how is this if condition activated?  Perhaps by passing something on the command line to cargo test ...